### PR TITLE
[MRG] Implementation of RAAA method for EOG correction

### PR DESCRIPTION
**Pull request by wmvanvliet**
I've been using this method of EOG correction for a few years now with great succes. Thought I'd share.

This is a method of removing EOG artefacts that does not rely on blind source separation. It performs a linear regression of the EOG channels onto the EEG (and/or MEG) and removes it. Like SSP and ICA, it has its own up- and downsides.

Normally you would use this with a small calibration session at the start of your experiment. You make the subject follow an icon across the screen to capture saccades and make the subject blink a few times. Then, you can construct clean epochs around these events and estimate the linear relationship between the EOG sensors and the rest of the signal. Bonus points if you also record rEOG. You can read the entire story here:

Croft, R. J., & Barry, R. J. (2000). Removal of ocular artifact from the
EEG: a review. Clinical Neurophysiology, 30(1), 5-19.
http://doi.org/10.1016/S0987-7053(00)00055-1

TODO:

 - [x] Write tests

### CODE REVIEW

**Comment by kingjr** [comment]
Nice! I wanted to test this for a while.

In principle you could accept epochs instances too, no?

**Comment by dengemann** [comment]
Small remark: maybe let's keep roughly the directory structure of MNE-Python, will make it easier in the future.

**Comment by Eric89GXL** [comment]
So put things roughly where we would want them to land in the production
package? +1 from me

**Comment by dengemann** [comment]
> So put things roughly where we would want them to land in the production
package?

yep.

**Comment by kingjr** [comment]
Ping me once travis and all are set for a review.

**Comment by wmvanvliet** [comment]
like this?

**Comment by dengemann** [comment]
yes good for now. we can then add some algorithms.py file somewhere to collect different peproc strategies.

File: mne_sandbox/preprocessing/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+from eog import eog_regression
```
Comment (agramfort) [inline mne_sandbox/preprocessing/__init__.py]: Relative import please

File: examples/preprocessing/eog_regression.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
+"""
```
Comment (agramfort) [inline examples/preprocessing/eog_regression.py]: Start example with plot

File: mne_sandbox/preprocessing/eog.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
+import numpy as np
+from numpy.linalg import lstsq
+from mne import pick_types
+
+def eog_regression(raw, blink_epochs, saccade_epochs=None, reog=None,
```
Comment (agramfort) [inline mne_sandbox/preprocessing/eog.py]: Pep8

**Comment by agramfort** [comment]
Can you copy the Travis config and add a proper setup.py ?

**Comment by wmvanvliet** [comment]
who shall I add as maintainer?

**Comment by wmvanvliet** [comment]
lets move the repository related chatter to Gitter.

**Comment by wmvanvliet** [comment]
This is now ready to be merged into the sandbox.

**Comment by agramfort** [comment]
Travis is not happy

**Comment by wmvanvliet** [comment]
I know, but it's not related to this PR. The test for this PR passes.

**Comment by dengemann** [comment]
@wmvanvliet are you happy? I think we should feel comfortable about developing a bit faster and more risky here, it's the sandbox.

**Comment by wmvanvliet** [comment]
hold on, got Travis to work... it actually detected a bug in my code :) @agramfort is a wise man.

**Comment by wmvanvliet** [comment]
Travis is happy. Feel free to merge.

**Comment by agramfort** [comment]
hold on, got Travis to work... it actually detected a bug in my code :) @agramfort is a wise man.

As long as you don't say old +1 :)

**Pull request merged**